### PR TITLE
[12.0][MIG] sale double validation

### PR DESCRIPTION
**Pull request by tschanzt**
based on #423 and #443
Moves the validation point to creation as we thought confirm is too late.

### CODE REVIEW

File: sale_double_validation/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+{
+    'name': 'Double validation for Sales',
+    'summary': "",
+    'version': '12.0.1.0.0',
+    'author': 'Camptocamp,Odoo Community Association (OCA)',
+    'maintainer': 'Camptocamp',
+    'license': 'AGPL-3',
+    'category': 'sale',
+    'depends': [
+        'sale',
```
Comment (nikul-serpentcs) [inline sale_double_validation/__manifest__.py:13]: @tschanzt no need to `sale`,  `sale_management` depends on `sale`

Comment (damdam-s) [reply sale_double_validation/__manifest__.py:13]: @nikul-serpentcs you're right. IMHO, I think explicit is always better than implicit :smile:

**Review by rousseldenis — APPROVED** [review]
Code review

File: sale_double_validation/__manifest__.py (hunk 1 above)
Comment (leemannd) [reply sale_double_validation/__manifest__.py:13]: Hello @nikul-serpentcs , with the answer from @damdam-s , is it ok for you?
If yes can you update your review?

**Comment by GSLabIt** [comment]
An info ..why approve_action set order to draft state when approved? I'd expect state = sale.

**Comment by tschanzt** [comment]
@GSLabIt The thought behind it was that if you create an offer for a client you want it to be checked before it is sent to the client so you can spot errors. To validate after the client has received the offer and accepted makes less sense for me.

**Comment by GSLabIt** [comment]
@tschanzt i think would make more sense to approve a sales order instead, that said, i'd change it on my local, or perhaps add a configuration to choose if approve quotation or sale order.

**Review by leemannd — APPROVED** [review]
Thanks

**Pull request merged**